### ISSUE TITLE
When I replace a node within transform, my callback then gets called for the new one

### ISSUE THREAD

**Issue by javazen** [josdejong/mathjs#1428]
In processing a parse tree, the callback that I pass to transform may replace SymbolNode "a" with a  newly created SymbolNode "p0". I was surprised to see that I then also get called for the new "p0" node. I can easily work around this, but is this expected?

**Comment by josdejong** [comment]
Thanks for reporting Sam. You're right that it's not so logical to iterate over replaced nodes. That could even result into infinite loops. It wasn't really on purpose though I've noticed the behavior before. 

I've adjusted the behavior, see https://github.com/josdejong/mathjs/commit/5c263d3dff9e74144d57f8ccfc7c7c919f40e6aa

We could call it a bug and do a minor bug fix release, but it is very well possible that existing code relies on the behavior (for example `math.simplify` relied on it too!), so I would like to move this change into the first next breaking release (v6.0.0).
